src/coreclr/tools/aot/ILCompiler.Compiler/Compiler/SubstitutionProvider.cs

```diff
@@ -73,18 +73,14 @@ private bool TryGetFeatureCheckValue(EcmaMethod method, out bool value)
                 if (featureGuardAttribute.FixedArguments is not [CustomAttributeTypedArgument<TypeDesc> { Value: EcmaType featureType }])
                     continue;
 
-                if (featureType.Namespace == "System.Diagnostics.CodeAnalysis") {
```

Comment:
This has more diffs than needed because I also fixed the formatting.

Do we have a plan for what to do with the illinker formatting? I don't mind illinker's charming odd formatting but I do care when the formatting leaks out. This is not the first or second or third time it leaked out.

We also leak it into places like learn.microsoft.com and can't even be consistent about it there. It makes things look sloppy.

Reply:
IMHO, we should just align the formatting to one standard repo-wide and fail at least one leg in CI matrix in PRs when this produces diffs: https://github.com/dotnet/runtime/blob/54ff0d270aea2b1904792e457970cdb96912ac18/eng/formatting/format.sh#L20

There are no points in adhering to "historical reasons" at this point. 🙂

Reply:
We very rarely squabble over formatting in this repo. I'm confident 90% of all formatting nits I ever had in PRs were with illinker devs doing non-illinker work.

There are times when people violate the formatting rules because it's done to highlight structure that would be lost if formatted according to the book. I'm pretty sure a formatting police would demotivate people from doing that because nobody wants to research how to pragma suppress the formatter (I would certainly quit doing that because I _hate_ pragma suppressions). The formatter would likely destroy more readability than what it would bring; I don't see formatting as a general problem in the PRs I review (or in reviews I get on my PRs).

Reply:
> fail at least one leg in CI matrix in PRs when this produces diffs:

There are already JIT formatting jobs with clang-format when you touch any files in src/coreclr/jit. Presumably the same thing can be done here if there's an agreement.

Reply:
> There are already JIT formatting jobs with clang-format when you touch any files in src/coreclr/jit. Presumably the same thing can be done here if there's an agreement.

You can enforce formatting via Roslyn analyzer. It has been enabled for libraries for quite some time. No need to have an formatting job.

Reply:
> > fail at least one leg in CI matrix in PRs when this produces diffs:
> 
> There are already JIT formatting jobs with clang-format when you touch any files in src/coreclr/jit. Presumably the same thing can be done here if there's an agreement.

Here's how they work (just the PRs I was pinged on in the past half a day or so):

![image](https://github.com/dotnet/runtime/assets/13110571/f2d71234-4915-48be-9fdd-005bc719482e)

![image](https://github.com/dotnet/runtime/assets/13110571/c403db20-c7ff-4342-81ef-bc2e18b48602)

They:
1. Annoy people working in the codebase
2. Produce an extra dose of CO2 (notice the :x:)
3. Could be a fix to this manufactured problem. Using different coding styles in different part of the product is an entirely manufactured problem; I don't remember when I had to comment on formatting (or fix it) outside interactions with _anyone_ exposed to illinker codebase; I think the only person who was able to maintain the separation was Marek because I don't recall issues with that but that would be an exception to a rule.

Reply:
> I don't remember when I had to comment on formatting (or fix it) outside interactions with anyone exposed to illinker codebase

Could be because libs subset already enforces the formatting rules and fails the build (seen a single trailing space failing builds a few times).

> Annoy people working in the codebase

Bit annoying it is, if we are in a rush. Which begs the question; why the rush merging PR which affects everyone? 🙂
I guess it is ok; better than annoying multiple people with unformatted code to save one developer's 5-10 seconds of time in active PR, who wasn't using VS or forgot to press `Ctrl+K,D` before `Ctrl+S`. They can use `dotnet format <changed files>` or the helper script in repo linked above (which only formats files in changeset, according to the .editorconfig and CodeAnalysis rules file).

Reply:
> Could be because libs subset already enforces the formatting rules and fails the build (seen a single trailing space failing builds a few times).

99% of code I write or review doesn't have build-enforced formatting. I rarely review/work in the libs or JIT subsets. It's not that. It's just not a problem in general.